gnome-initial-setup/pages/network/gis-network-page.c

```diff
@@ -588,11 +594,48 @@ active_connections_changed (NMClient *client, GParamSpec *pspec, GisNetworkPage
   refresh_wireless_list (page);
 }
 
+static gboolean
+skip_page_for_ethernet_connection (GisNetworkPage  *self,
+                                   GNetworkMonitor *monitor)
+{
+  GisNetworkPagePrivate *priv;
+  GisAssistant *assistant;
+  gboolean is_current_page;
+
+  priv = gis_network_page_get_instance_private (self);
+  assistant = gis_driver_get_assistant (GIS_PAGE (self)->driver);
+  is_current_page = gis_assistant_get_current_page (assistant) == GIS_PAGE (self);
+
+  if (!is_current_page ||
+      priv->row_clicked ||
+      g_network_monitor_get_connectivity (monitor) != G_NETWORK_CONNECTIVITY_FULL)
+    {
+      return FALSE;
+    }
+
+
+  gtk_widget_hide (GTK_WIDGET (self));
+
+  if (is_current_page)
+    gis_assistant_next_page (gis_driver_get_assistant (GIS_PAGE (self)->driver));
```

Comment:
I don't think this is right; we never want to change the UI without user interaction. If the connectivity happens to change to full while the page is visible, we should just make the "Next" button active.